blockchain/validators/src/metadata/avr/config.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Configuration for the avr history bootstrap file.
+
+use crate::error::ParseError;
+
+use mc_blockchain_types::{BlockIndex, VerificationReport, VerificationSignature};
+use mc_common::ResponderId;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde_with::{hex, serde_as, DeserializeAs, SerializeAs};
+use std::{fs, option::Option, path::Path};
+
+/// Struct for reading historical Intel Attestation Verification Report
+/// (AVR) data from a configuration file.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct AvrHistoryConfig {
+    /// List of AvrHistoryRecord objects sorted by ResponderId and block range
+    pub node: Vec<AvrHistoryRecord>,
+}
+
+/// Stores a historical AVR record (or lack thereof) for a given
+/// [ResponderId] and block range
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, Eq, Ord, PartialEq, PartialOrd, Serialize)]
+pub struct AvrHistoryRecord {
+    /// Uri of the consensus node
+    pub responder_id: ResponderId,
+
+    /// Block the AVR Report for the signing key becomes valid
+    pub first_block_index: BlockIndex,
+
+    /// Final block the AVR Report for the signing key is valid
+    pub last_block_index: Option<BlockIndex>,
+
+    /// AVR Report (or lack thereof) for the node & block ranges
+    #[serde_as(as = "Option<VerificationReportShadow>")]
+    #[serde(default)]
+    pub avr: Option<VerificationReport>,
+}
+
+impl AvrHistoryConfig {
+    /// Load the [AvrHistoryConfig] from a .json or .toml file
+    ///
+    /// # Arguments
+    ///
+    /// * `path` - Path to the configuration file containing
+    /// the history of AVRs generated by MobileCoin consensus node
+    /// enclaves
+    pub fn load(path: impl AsRef<Path>) -> Result<AvrHistoryConfig, ParseError> {
```

Comment:
nit: one can use `Self` to minimize type name changes
```suggestion
    pub fn load(path: impl AsRef<Path>) -> Result<Self, ParseError> {
```